R/robustbase-glmrob-tidiers.R

```diff
@@ -0,0 +1,114 @@
+#' @templateVar class glmrob
+#' @template title_desc_tidy_lm_wrapper
+#'
+#' @param x A `glmrob` object returned from [robustbase::glmrob()].
+#' 
+#' @details For tidiers for robust models from the \pkg{MASS} package see
+#'   [tidy.rlm()]. For tidiers for robust models from the \pkg{robust} package
+#'   see [tidy.lmRob()].
+#'
+#' @inherit tidy.lmrob examples
+#'
+#' @export
+#' @family robustbase tidiers
+#' @rdname tidy.robustbase.glmrob
+#' @seealso [robustbase::glmrob()]
+#' @include stats-lm-tidiers.R
+tidy.glmrob <- function (x, ...) {
+  dots <- rlang::enquos(...)
+  dots$conf.int <- FALSE
+  
+  ret <- rlang::exec(tidy.lm, x, !!!dots)
+  
+  # tidy.glmrob(..., conf.int = TRUE) throws an error, 
+  # so calc conf.int separately if requested
+  conf.int <- rlang::`%||%`(rlang::dots_list(...)$conf.int, FALSE)
+  
+  if (conf.int) {
+    level <- rlang::`%||%`(rlang::dots_list(...)$level, 0.95)
+    ci <- stats::confint.default(x, level = level) %>% 
+      as_tibble(rownames = NA)  %>% 
+      tibble::rownames_to_column(var = 'term')
+    names(ci)[2:3] <- c("conf.low", "conf.high")
+
+    ret <- ret %>% 
+      left_join(ci, by = 'term')
+  }
+  
+  ret
+}
+
+
+#' @templateVar class glmrob
+#' @template title_desc_augment
+#' 
+#' @inherit tidy.glmrob params 
+#' @template param_data
+#' @template param_newdata
+#' 
+#' @param se_fit a switch indicating if standard errors are required.
+#' 
+#' @details For tidiers for robust models from the \pkg{MASS} package see
+#'   [tidy.rlm()]. For tidiers for robust models from the \pkg{robust} package
+#'   see [tidy.lmRob()].
+#'
+#' @export
+#' @family robustbase tidiers
+#' @rdname augment.robustbase.glmrob
+#' @seealso [robustbase::glmrob()]
+#' @include stats-lm-tidiers.R
+augment.glmrob <- function(x, data = model.frame(x), newdata = NULL,
+                           type.predict = c("link", "response", "terms"),
+                           type.residuals = c("deviance", "pearson"),
+                           se_fit = FALSE, ...) {
+  # slightly modified version of `augment.glm`
+  type.predict <- match.arg(type.predict)
```

Comment:
Let's use the `rlang` version here as it gives a better error message.

```suggestion
  type.predict <- arg_match(type.predict)
```